src/typedarray.rs

```diff
@@ -132,13 +132,20 @@ impl<'a, T: TypedArrayElementCreator + TypedArrayElement> TypedArray<'a, T> {
             return Err(());
         }
 
+        TypedArray::<T>::update(length, data, result);
+        
+        Ok(())
+    }
+    
+    ///  Update an existed JS typed array
+    pub unsafe fn update(length: u32,
+                         data: Option<&[T::Element]>,
+                         result: MutableHandleObject) {
```

Comment:
I argue that we can create a update_raw that encapsulates the relevant bit, and call that from both create and update (passing in a mutable destination buffer and an immutable source buffer).